CODEWALK.md

```diff
@@ -302,6 +308,58 @@ A null block count indicates the absence of any blocks mined in a previous round
 Subsequent blocks are built upon *all* of the tipset; 
 there is a canonical ordering of the messages in a tipset defining a new consensus state, not directly referenced from any of the tipset’s blocks.
 
+### Storage protocol
+The storage protocol is mechanism by which clients make deals directly with storage miners to store their data, implemented in [`protocol/storage`](https://github.com/filecoin-project/go-filecoin/blob/master/protocol/storage).
+
+A storage miner ([protocol/storage/miner.go](https://github.com/filecoin-project/go-filecoin/blob/master/protocol/storage/miner.go)) advertises storage with an on-chain ask, 
+which specifies an asking price and storage capacity at that price. 
+Clients discover asks by iterating miner actors’ on-chain state. 
+A client wishing to take an ask creates a deal proposal. 
+A proposal references a specific unit of data, termed a piece, which has a CID (hash of the bytes). 
+A piece must fit inside a single sector (see below) as defined by network parameters.
+
+A storage client ([protocol/storage/client.go](https://github.com/filecoin-project/go-filecoin/blob/master/protocol/storage/client.go)) connects directly to a miner to propose a deal, 
+using a libp2p peer id embedded in the on-chain storage miner actor data. 
+An off-chain lookup service maps peer ids to concrete addresses, in the form of multiaddr, using a [libp2p distributed hash table](https://github.com/filecoin-project/go-filecoin/blob/master/networking.md) (DHT). 
+A client also creates a payment channel so the miner can be paid over time for storing the piece. 
+The miner responds with acceptance or otherwise.
+
+When proposing a deal, a client loads the piece data into its [IPFS block service](https://github.com/ipfs/go-blockservice). 
+This advertises the availability of the data to the network, making it available to miners. 
+A miner accepting a deal pulls the data from the client (or any other host) using the IPFS block service [bitswap protocol](https://github.com/ipfs/specs/tree/master/bitswap).
```

Comment:
@whyrusleeping could you confirm or correct our understanding here?

Reply:
this is currently what is happening, however we will be changing this in the near future, to at least just use bitswap to pull from a single peer, depending on how graphsync work goes, we may use graphsync (bitswap v2 basically) to fetch the data.

Reply:
this also depends on the resolution of this issue: https://github.com/filecoin-project/specs/issues/136